cl/users/views.py

```diff
@@ -613,26 +615,40 @@ def password_change(request: AuthenticatedHttpRequest) -> HttpResponse:
 
 
 @csrf_exempt  # nosemgrep
-def mailchimp_webhook(request: HttpRequest) -> HttpResponse:
-    """Respond to changes to our mailing list"""
-    logger.info("Got mailchimp webhook with %s method.", request.method)
+def moosend_webhook(request: HttpRequest) -> HttpResponse:
+    logger.info("Got moosend webhook with %s method.", request.method)
+
     if request.method == "POST":
-        webhook_type = request.POST.get("type")
-        logger.info("Handling mailchimp '%s' request.", webhook_type)
-        wants_newsletter = None
-        if webhook_type == "subscribe":
-            wants_newsletter = True
-        elif webhook_type == "unsubscribe":
-            wants_newsletter = False
-        if wants_newsletter is not None:
-            # Only update this value if we get a valid webhook_type
-            email = request.POST.get("data[email]")
-            profiles = UserProfile.objects.filter(user__email=email)
-            logger.info(
-                "Updating %s profiles for email %s", profiles.count(), email
+        # The body is returned as a byte string
+        body = request.body.decode("utf-8")
```

Comment:
Thanks for the suggestion, I'm not 100% sure but I think I can't use requests.text here because I'm getting Django's HttpRequest object when Moosend calls this webhook instead a Request object when using python requests.

Reply:
Oh, right!